Mathlib/NumberTheory/LSeries/RiemannZeta.lean

```diff
@@ -0,0 +1,246 @@
+  hurwitzZetaEven_neg_two_mul_nat_add_one 0 n
+#align riemann_zeta_neg_two_mul_nat_add_one riemannZeta_neg_two_mul_nat_add_one
+
+/-- Riemann zeta functional equation, formulated for `ζ`: if `1 - s ∉ ℕ`, then we have
+`ζ (1 - s) = 2 ^ (1 - s) * π ^ (-s) * Γ s * sin (π * (1 - s) / 2) * ζ s`. -/
+theorem riemannZeta_one_sub {s : ℂ} (hs : ∀ n : ℕ, s ≠ -n) (hs' : s ≠ 1) :
+    riemannZeta (1 - s) = 2 * (2 * π) ^ (-s) * Gamma s * cos (π * s / 2) * riemannZeta s := by
+  rw [riemannZeta, hurwitzZetaEven_one_sub 0 hs (Or.inr hs'), cosZeta_zero, hurwitzZetaEven_zero]
+#align riemann_zeta_one_sub riemannZeta_one_sub
+
+/-- A formal statement of the **Riemann hypothesis** – constructing a term of this type is worth a
+million dollars. -/
+def RiemannHypothesis : Prop :=
+  ∀ (s : ℂ) (_ : riemannZeta s = 0) (_ : ¬∃ n : ℕ, s = -2 * (n + 1)) (_ : s ≠ 1), s.re = 1 / 2
+#align riemann_hypothesis RiemannHypothesis
+
+/-!
+## Relating the Mellin transform to the Dirichlet series
+-/
+
+theorem completedZeta_eq_tsum_of_one_lt_re {s : ℂ} (hs : 1 < re s) :
+    completedRiemannZeta s =
+      (π : ℂ) ^ (-s / 2) * Gamma (s / 2) * ∑' n : ℕ, 1 / (n : ℂ) ^ s := by
+  have := (hasSum_nat_completedCosZeta 0 hs).tsum_eq.symm
+  simp only [QuotientAddGroup.mk_zero, completedCosZeta_zero] at this
+  simp only [this, Gammaℝ_def, mul_zero, zero_mul, Real.cos_zero, ofReal_one, mul_one, mul_one_div,
+    ← tsum_mul_left]
+  congr 1 with n
+  split_ifs with h <;>
+  simp only [h, Nat.cast_zero, zero_cpow (Complex.ne_zero_of_one_lt_re hs), div_zero, mul_zero]
+#align completed_zeta_eq_tsum_of_one_lt_re completedZeta_eq_tsum_of_one_lt_re
+
+/-- The Riemann zeta function agrees with the naive Dirichlet-series definition when the latter
+converges. (Note that this is false without the assumption: when `re s ≤ 1` the sum is divergent,
+and we use a different definition to obtain the analytic continuation to all `s`.) -/
+theorem zeta_eq_tsum_one_div_nat_cpow {s : ℂ} (hs : 1 < re s) :
+    riemannZeta s = ∑' n : ℕ, 1 / (n : ℂ) ^ s := by
+  simpa only [QuotientAddGroup.mk_zero, cosZeta_zero, mul_zero, zero_mul, Real.cos_zero,
+    ofReal_one] using (hasSum_nat_cosZeta 0 hs).tsum_eq.symm
+#align zeta_eq_tsum_one_div_nat_cpow zeta_eq_tsum_one_div_nat_cpow
+
+/-- Alternate formulation of `zeta_eq_tsum_one_div_nat_cpow` with a `+ 1` (to avoid relying
+on mathlib's conventions for `0 ^ s`).  -/
+theorem zeta_eq_tsum_one_div_nat_add_one_cpow {s : ℂ} (hs : 1 < re s) :
+    riemannZeta s = ∑' n : ℕ, 1 / (n + 1 : ℂ) ^ s := by
+  have hs' : s ≠ 0 := fun h ↦ (not_lt.mpr zero_le_one) (zero_re ▸ h ▸ hs)
+  have := zeta_eq_tsum_one_div_nat_cpow hs
+  rw [tsum_eq_zero_add] at this
+  · simpa [Nat.cast_zero, zero_cpow hs', div_zero, zero_add, Nat.cast_add, Nat.cast_one]
+  · refine .of_norm ?_
+    simp_rw [norm_div, norm_one, Complex.norm_eq_abs, ← ofReal_natCast,
+      abs_cpow_eq_rpow_re_of_nonneg (Nat.cast_nonneg _) (zero_lt_one.trans hs).ne',
+      summable_one_div_nat_rpow]
```

Comment:
Variants of this are already present (e.g., [LSeriesSummable_of_bounded_of_one_lt_re](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/LSeries/Basic.html#LSeriesSummable_of_bounded_of_one_lt_re)), or, perhaps closer, [summable_riemannZetaSummand](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/EulerProduct/DirichletLSeries.html#summable_riemannZetaSummand)). Still, I don't think the statement is available in this precise form. It might be a consideration to add it next to [Real.summable_one_div_nat_rpow](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/PSeries.html#Real.summable_one_div_nat_rpow).

Reply:
Just to be clear: you're suggesting we add something like `Complex.summable_one_div_nat_cpow`?

Reply:
Done. I made a new file for it in order to avoid massively increasing the imports of `Analysis.PSeries`.